pxtlib/auth.ts

```diff
@@ -674,7 +684,7 @@ namespace pxt.auth {
         const hash = callbackState.hash.startsWith('#') ? callbackState.hash : `#${callbackState.hash}`;
         const params = pxt.Util.stringifyQueryString('', callbackState.params);
         const pathname = loginState.callbackPathname.startsWith('/') ? loginState.callbackPathname : `/${loginState.callbackPathname}`;
-        const redirect = `${pathname}${hash}${params}`;
+        const redirect = `${pathname}${params}${hash}`;
```

Comment:
Oh, this is a nice tiny side fix :)

Reply:
What did this fix exactly?

Reply:
in a correctly formatted url parameters (`?abc=1`) come before the has (`tutorial:ajskdlf`) -- if it is set after the hash they are interpreted as part of the hash. e.g.:

<img width="204" alt="image" src="https://user-images.githubusercontent.com/5615930/228886583-35cedf57-c18b-4326-90d9-ba45833555ab.png">

vs

<img width="210" alt="image" src="https://user-images.githubusercontent.com/5615930/228886735-914eb9c5-fbff-4a67-b8c0-771d63322e4f.png">

In a lot of locations we handle this as it's easy for users to make mistakes when constructing the url (e.g. using a regex on `window.location` instead of `URLSearchParams` to check when a parameter is set) but it's nice to keep it correctly formatted url when possible.